File: stop_processes_on_port.cs
using System;

class StopProcesses
{
    static void Main()
    {
        Console.Write("Enter the port number: ");
        string port = Console.ReadLine().Trim();

        if (int.TryParse(port, out int portNumber))
        {
            System.Diagnostics.Process.Start("taskkill", $"/F /PID {portNumber}");
            Console.WriteLine($"Processes using port {port} have been terminated.");
        }
        else
        {
            Console.WriteLine("Invalid port number. Please enter a valid numeric port.");
        }

        Console.WriteLine("Press Enter to exit.");
        Console.ReadLine();
    }
}

File: stop_processes_on_port.rb
print "Enter the port number: "
port = gets.chomp

if port.match?(/^\d+$/)
  system("kill -9 $(lsof -t -i:#{port})")
  puts "Processes using port #{port} have been terminated."
else
  puts "Invalid port number. Please enter a valid numeric port."
end

print "Press Enter to exit"
gets
